### ISSUE TITLE
grpc_coverage broken, missing python3.7

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#17198]
Consistently red starting from https://source.cloud.google.com/results/invocations/b8315ded-a38c-4fc7-9d46-7ea252566544/targets

Probably after merging https://github.com/grpc/grpc/pull/16995

```
2018-11-07 19:33:47,859 ++ dirname /var/local/git/grpc/tools/run_tests/helper_scripts/build_python.sh
+ cd /var/local/git/grpc/tools/run_tests/helper_scripts/../../..
++ uname -s
+ PLATFORM=Linux
+ PYTHON=python3.7
+ VENV=py37_native
+ VENV_RELATIVE_PYTHON=bin/python
+ TOOLCHAIN=unix
++ is_msys
++ '[' Linux '!=' Linux ']'
++ exit 1
+ '[' '' ']'
++ pwd
+ ROOT=/var/local/git/grpc
+ export 'CFLAGS=-I/var/local/git/grpc/include -std=gnu99 -fno-wrapv '
+ CFLAGS='-I/var/local/git/grpc/include -std=gnu99 -fno-wrapv '
+ export GRPC_PYTHON_BUILD_WITH_CYTHON=1
+ GRPC_PYTHON_BUILD_WITH_CYTHON=1
+ export LANG=en_US.UTF-8
+ LANG=en_US.UTF-8
+ export GRPC_PYTHON_BUILD_EXT_COMPILER_JOBS=4
+ GRPC_PYTHON_BUILD_EXT_COMPILER_JOBS=4
++ is_linux
++ '[' '' '!=' Linux ']'
++ echo true
+ '[' true ']'
++ command -v ccache
+ '[' -x /usr/bin/ccache ']'
++ command -v gcc
+ '[' -x /usr/bin/gcc ']'
+ export 'CC=ccache gcc'
+ CC='ccache gcc'
+ python3.7 -m pip install --user virtualenv
/var/local/git/grpc/tools/run_tests/helper_scripts/build_python.sh: line 138: python3.7: command not found
2018-11-07 19:33:47,859 FAILED: /var/local/git/grpc/tools/run_tests/helper_scripts/build_python.sh [ret=127, pid=3191, time=0.0sec]
```

**Comment by jtattermusch** [comment]
CC @lidizheng 

this is the dockerfile being used for coverage: https://github.com/grpc/grpc/blob/master/tools/dockerfile/test/multilang_jessie_x64/Dockerfile

**Comment by lidizheng** [comment]
I started to investigate this problem two days ago. And I went into various pitfalls.

1. The dependency conflict of `binutil` between Debian `testing` and Debian `jessie`. And it results to unable to install Python 3.7 in the most easy and robust way. 
2. The `.NET Core 1.0.0` doesn't like `debian:stretch`. And it prevented me from upgrade the whole docker image to `stretch`.
3. The ssl compilation of Python 3.7 in `jessie` is a pain. [CPython#34028](https://bugs.python.org/issue34028)
